zproject/backends.py

```diff
@@ -1212,7 +1218,7 @@ def social_auth_finish(backend: Any,
     # subdomain.zulip.example.com that will verify the signature and
     # then call login_or_register_remote_user.
     return redirect_and_log_into_subdomain(
-        realm, full_name, email_address,
+        realm, full_name, email_address, user_avatar_url,
```

Comment:
With the absurd amount of arguments that these functions take, I think it may be be best to use `user_avatar_url` always as a keyword argument, rather than positional - verifying that there isn't a bug caused by getting the order of positional arguments wrong is not fun :confounded:

Reply:
ok. I was actually thinking if it would be nicer if the auth backend passed an object instead of all these attributes. Because every time I want to use something new, like user_avatar_url,  I need to pass it down to a loooot of functions, while if there was an object, I would just add that attribute once. Of course not to implement this on this PR but maybe it would be a cool refactor.

Reply:
Agreed, that's one of the things that https://github.com/zulip/zulip/pull/14015 is meant to do